app/pages/app/app.js

```diff
@@ -343,6 +344,34 @@ export class AppComponent extends React.Component {
     return null;
   }
 
+  renderNavPatientHeader() {
+    const { 
+      patient, 
+      user,
+      permsOfLoggedInUser,
+      context: { trackMetric },
+      clinicFlowActive,
+      selectedClinicId,
+      location: currentPage,
+      query,
+    } = this.props;
+
+    if (!this.isPatientVisibleInNavbar()) return null; // only show on pages with a patient of focus
+    
+    return (
+      <NavPatientHeader 
+        patient={patient} 
+        user={user} 
+        trackMetric={trackMetric} 
+        permsOfLoggedInUser={permsOfLoggedInUser}
+        clinicFlowActive={clinicFlowActive}
+        selectedClinicId={selectedClinicId}
+        currentPage={currentPage}
+        query={query}
+      />
```

Comment:
I have some ideas that would help us reduce prop drilling and help us decouple our components quite a bit. I think would be quite simple and painless in our case. This PR is a little big so I won't suggest them here, but I would love to chat about it.